lib/ocpp/v201/charge_point.cpp

```diff
@@ -27,31 +27,27 @@ bool Callbacks::all_callbacks_valid() const {
 ChargePoint::ChargePoint(const std::map<int32_t, int32_t>& evse_connector_structure,
                          const std::string& device_model_storage_address, const std::string& ocpp_main_path,
                          const std::string& core_database_path, const std::string& sql_init_path,
-                         const std::string& message_log_path, const std::string& certs_path,
-                         const Callbacks& callbacks) :
-    ocpp::ChargingStationBase(),
+                         const std::string& message_log_path, const std::shared_ptr<EvseSecurity> evse_security,
+                         const Callbacks& callbacks,
+                         const std::optional<SecurityConfiguration> security_configuration) :
+    ocpp::ChargingStationBase(evse_security, security_configuration),
     registration_status(RegistrationStatusEnum::Rejected),
     websocket_connection_status(WebsocketConnectionStatusEnum::Disconnected),
     operational_state(OperationalStatusEnum::Operative),
     network_configuration_priority(0),
     disable_automatic_websocket_reconnects(false),
     reset_scheduled(false),
     reset_scheduled_evseids{},
-    callbacks(callbacks),
     firmware_status(FirmwareStatusEnum::Idle),
-    upload_log_status(UploadLogStatusEnum::Idle) {
-
+    upload_log_status(UploadLogStatusEnum::Idle),
+    callbacks(callbacks) {
     // Make sure the received callback struct is completely filled early before we actually start running
     if (!this->callbacks.all_callbacks_valid()) {
         EVLOG_AND_THROW(std::invalid_argument("All non-optional callbacks must be supplied"));
     }
-
     this->device_model = std::make_unique<DeviceModel>(device_model_storage_address);
-    this->pki_handler = std::make_shared<ocpp::PkiHandler>(
-        certs_path,
-        this->device_model->get_optional_value<bool>(ControllerComponentVariables::AdditionalRootCertificateCheck)
-            .value_or(false));
-    this->database_handler = std::make_shared<DatabaseHandler>(core_database_path, sql_init_path);
+    this->evse_security = evse_security;
```

Comment:
Please remove this line. evse_security is already initialized in ChargingStationBase. If you set evse_security to nullptr, a new instance is created in ChargingStationBase and overwritten with this line.